nrf-hal-common/src/uarte.rs

```diff
@@ -135,31 +201,26 @@ where
             unsafe { w.bits(1) });
 
         // Wait for transmission to end.
-        let mut endtx;
-        let mut txstopped;
-        loop {
-            endtx = self.0.events_endtx.read().bits() != 0;
-            txstopped = self.0.events_txstopped.read().bits() != 0;
-            if endtx || txstopped {
-                break;
-            }
+        while self.0.events_endtx.read().bits() == 0 {
```

Comment:
the `events_txstopped.reset();` above is no longer needed.

Reply:
Noted. Will remove.